libs/ml/tests/unit/graph/graph.cpp

```diff
@@ -369,25 +369,25 @@ TYPED_TEST(GraphTest, variable_freezing_subgraph)
   auto gradient_vector = g.GetGradients();
   for (auto grad : gradient_vector)
   {
-    grad.Fill(static_cast<DataType>(2.0));
+    grad.Fill(fetch::math::Type<DataType>("2.0"));
   }
   g.ApplyGradients(gradient_vector);
 
   // Get weights after applying gradient
   auto weights_2 = g.GetWeights();
 
-  ASSERT_TRUE(weights_1.at(0).AllClose(weights_2.at(0), static_cast<DataType>(0),
-                                       static_cast<DataType>(0)));
-  ASSERT_TRUE(weights_1.at(1).AllClose(weights_2.at(1), static_cast<DataType>(0),
-                                       static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_1.at(2).AllClose(weights_2.at(2), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_1.at(3).AllClose(weights_2.at(3), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_TRUE(weights_1.at(4).AllClose(weights_2.at(4), static_cast<DataType>(0),
-                                       static_cast<DataType>(0)));
-  ASSERT_TRUE(weights_1.at(5).AllClose(weights_2.at(5), static_cast<DataType>(0),
-                                       static_cast<DataType>(0)));
+  ASSERT_TRUE(weights_1.at(0).AllClose(weights_2.at(0), fetch::math::Type<DataType>("0"),
```

Comment:
(optional) DataType{0} could be a bit more clear here, especially as there are only integers in this code block